docs/schema_plugins.md

```diff
@@ -1 +1,234 @@
+     \-- author
+     |    +-- name: String[]
+     |    +-- (email: String[])
+     \-- creator
+     |    +-- name: String[]
+     |    +-- version: String[]
+     \-- program
+          +-- name: String[]
+          +-- version: String[]
+
+The program information can be stored under the existing `Program()` class within `nomad-simulations`. We already learned how to implement this into our parser:
+
+```python
+from nomad_simulations.schema_packages.general import Simulation, Program
+
+simulation = Simulation()
+simulation.program = Program(
+    name=group_h5md_dict.get('program_name'),
+    version=group_h5md_dict.get('program_version'),
+)
+```
+
+ However, we need to create a new class for the rest of the metadata. The "creator" information is referring to a secondary software used to create the hdf5 file, so it makes sense to store this in a class very similar to `Program()`. Let's first take a look at the current `Program()` schema:
+
+```python
+from nomad_simulations.general import Program
+program = Program()
+print(program.m_def.all_quantities)
+```
+
+```
+{'name': nomad_simulations.general.Program.name:Quantity,
+ 'datetime': nomad.datamodel.metainfo.basesections.BaseSection.datetime:Quantity,
+ 'lab_id': nomad.datamodel.metainfo.basesections.BaseSection.lab_id:Quantity,
+ 'description': nomad.datamodel.metainfo.basesections.BaseSection.description:Quantity,
+ 'version': nomad_simulations.general.Program.version:Quantity,
+ 'link': nomad_simulations.general.Program.link:Quantity,
+ 'version_internal': nomad_simulations.general.Program.version_internal:Quantity,
+ 'compilation_host': nomad_simulations.general.Program.compilation_host:Quantity}
+ ```
+
+ Indeed, this class contains `name` and `version` information, along with other quantities that make sense in the context of the H5MD-NOMAD creator. So, in this case we can simply reuse this class. We need to simply add a new section within the `Simulation()` class for the H5MD-NOMAD file creator that is of type `Program()`. In `schema_packages/<parser_name>_schema.py`:
+
+ ```python
+class Simulation(nomad_simulations.schema_packages.general.Simulation):
+
+    x_h5md_creator = SubSection(
+        sub_section=nomad_simulations.schema_packages.general.Program.m_def
+    )
+```
+
+Here, we create a new class called `Simulation()` that inherits from `Simulation()` from `nomad-simulations` while adding the additional `x_h5md_creator` section of type `Program()`. Note that the `x_<program_name>_<quantity_name>` notation is our standard notation for specifying program-specific quantities that are not necessarily generalizable.
+
+Now we can use this new class within our parser to store the creator information:
+
+```python
+from nomad_simulations.schema_packages.general import Program
+from nomad_simulations.schema_packages/<parser_name>_schema.py import Simulation
+
+simulation = Simulation()
+simulation.x_h5md_creator = Program(
+    name=group_h5md_dict.get('h5md_creator_name'),
```

Comment:
Maybe I can add sometgin about this in the parser section: how to interface with HDF5 files?
I could briefly mention it and refer to here for an example.